### ISSUE TITLE
acf pro image

### ISSUE THREAD

**Issue by Karuma** [magicstickuk/multisite-post-duplicator#48]
Hi
I have a problem with duplicating images with ACF Pro image and gallery. I have a clean installation of WordPress multisite (subfolder) with 3 pages, ACF Pro and MPD on VVV. When I create copies of a post, only the images attached to the content are copied, Those added in the fields (Image / Gallery) do not copy. Cells in the database _postmeta are only a copy of 1:1

**Comment by Karuma** [comment]
oh i see. It dont work with acf fields hardcoded in template. Even if both sites use same template. ;/ is ther any hope for this to work good in both variants ?

**Comment by magicstickuk** [comment]
Is should work for both variants but only if the names (and keys in some instances) of the fields should match in both source and destination sites.

**Comment by Karuma** [comment]
Unfortunately, I checked it on a (normal) server with a fresh installation of WordPress and only 2 plugins MPD and ACF PRO. The fields that are entered in the template only work as long as they are basic fields. Image boxes and Galleries are simply copied as a string. However, when I added an new acf container (in ACF Builder) with Gallery and image to the page and then I added acf-multisite-sync and synchronized this field, Galleries and images were uploaded correctly.

**Comment by Karuma** [comment]
Hi 
If I understand correctly the acf-function.php file in line 40, you more or less check if such a field is acf and is it image gallery etc. Its oki but hardcoded metabox dont showup in tabel like this :(.

**Comment by Karuma** [comment]
There is a simple way to do this. Maybe it is not perfect but it is an option for admins and developers. Just as you have fields of excluded fields, you can do fields for the metaboxs programmed into the template. Gallery, Image and File.
foreach ($meta_values as $key => $meta) {

                //Indicates it could be a ACF Value
                if (isset($meta_values["_" . $key])) {

                    if (in_array($key, $acf_image_names)) {
                        $resultAdd = 'image';
                    } elseif (in_array($key, $acf_gallery_names)) {
                        $resultAdd = 'gallery';
                    } else {
                        global $wpdb;

                        $acf_field_key = $meta_values["_" . $key][0];

                        $tablename = mpd_get_tablename($wpdb->blogid);
                        $query = $wpdb->prepare("
                        SELECT post_content
                        FROM $tablename 
                        WHERE post_name = '%s'
                        AND post_type = 'acf-field'", $acf_field_key
                        );
                        //Get the posssible ACF controller post for this image
                        $result = $wpdb->get_row($query);
                    }
                    if ($result || $resultAdd) {

                        if (current_filter() == 'mpd_during_core_in_source') {
                            do_action('mpd_acf_field_found', $result, $meta, $acf_field_key, $destination_blog_id);
                        }
                        if ($result) {
                            $acf_control = unserialize($result->post_content);
                            $acf_type = $acf_control['type'];
                        }
                        if ($resultAdd) {
                            $acf_type = $resultAdd;
                        }

sorry for style it was just for test

**Comment by dominicpedro** [comment]
I can confirm this issue, and have been able to find a simple solution. 

If you register fields with php, there is no entry in the posts table to look up, same if you use the local json without sync.

By changing the acf lookup to use the function "get_field_object()" instead of looking up the field in the database posts table. Using the "$acf_field_key" in the function will return the field object with the field definition AND all values correctly formatted for that post.

Switching to this correctly copies over the images. 
I've not tested with the file or gallery, but cant see why it would not also work for them too.

example code: acf-functions.php line 40-odd...

```php
         $field = get_field_object($acf_field_key);

          if ($field) {
            $acf_type = $field['type'];

            switch ($acf_type) {
              case 'image':

                $acf_image_data_source = [
                  'image_id'      => $field['value']['ID'],
                  'field'         => $key,
                  'post_id'       => $post_id_to_copy,
                  'img_url'       => $field['value']['url'],
                  'img_post_mime' => $field['value']['mime_type']
                ];

                array_push($acf_collected, $acf_image_data_source);

                break;
              default:

                break;

            }
          } 
```

**Comment by dominicpedro** [comment]
any progress on this fix?

**Comment by magicstickuk** [comment]
this has now been fixed as part of pull request #53
